### PR TITLE
Add timeout

### PR DESCRIPTION
**Pull request by diclophis**

### CODE REVIEW

**Comment by diclophis** [comment]
could use a review here

File: test/unit/count_test.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ class CountTest < Test::Unit::TestCase
       should "count all documents by the leaving index empty" do
         c = Search::Count.new
         assert c.indices.empty?, "#{c.indices.inspect} should be empty"
-        assert_match %r|localhost:9200/_count|, c.url
+        assert_match %r|foobarbaz:9200/_count|, c.url
```
Comment (id4ho) [inline test/unit/count_test.rb:18]: do we want this? should we change it to `ENV["ELASTICSEARCH_URL"]`?

Comment (id4ho) [reply test/unit/count_test.rb:18]: Also, where does this foobarbaz env var come from? is that set in bigmaven?

File: lib/tire/http/client.rb [hunk 2]
```diff
@@ -8,39 +8,39 @@ class RestClient
         ConnectionExceptions = [::RestClient::ServerBrokeConnection, ::RestClient::RequestTimeout]
 
         def self.get(url, data=nil)
-          perform ::RestClient::Request.new(:method => :get, :url => url, :payload => data).execute
+          perform ::RestClient::Request.new(:method => :get, :url => url, :timeout => Configuration.timeout, :open_timeout => Configuration.timeout, :payload => data).execute
```
Comment (id4ho) [inline lib/tire/http/client.rb:11]: Maybe we should extract a method 
```
def perform_request(method:, url:, data:)
  perform ::RestClient::Request.new(
    :method => method,
    :url => url,
    :timeout => Configuration.timeout,
    :open_timeout => Configuration.timeout,
    :payload => data
  ).execute
end
```

Then we only need to change these in one place going forward.

File: Rakefile [hunk 3]
```diff
@@ -7,8 +7,8 @@ require 'rake/testtask'
 Rake::TestTask.new(:test) do |test|
   test.libs << 'lib' << 'test'
   test.test_files = FileList['test/unit/*_test.rb', 'test/integration/*_test.rb']
-  test.verbose = true
-  # test.warning = true
+  test.verbose = false
+  test.warning = false
```
Comment (id4ho) [inline Rakefile:11]: Was this necessary or was this part of you trying to get the specs green?

**Review by id4ho — COMMENTED** [review]
couple suggestions/questions

File: Rakefile (hunk 3 above)
Comment (diclophis) [reply Rakefile:11]: just trying to green up the specs

File: test/unit/count_test.rb (hunk 1 above)
Comment (diclophis) [reply test/unit/count_test.rb:18]: yea its part of me trying to fix the test suite

**Comment by diclophis** [comment]
I am going to merge as in, so we dont have to wait for another full re-index before we can get this into master bigmaven... then we can finish the test suite and make the refactor happen in a second branch

**Pull request merged**